File: e2e/java/src/test/java/dev/kreuzberg/e2e/BatchTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: batch. */
class BatchTest {
    @Test
    void testScrapeBatchBasic() throws Exception {
        // Batch scrape of multiple URLs all succeeding
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'batch.completed_count' not available on result type
        // skipped: field 'batch.failed_count' not available on result type
        // skipped: field 'batch.total_count' not available on result type
    }

    @Test
    void testScrapeBatchPartialFailure() throws Exception {
        // Batch scrape with one URL failing returns partial results
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'batch.completed_count' not available on result type
        // skipped: field 'batch.failed_count' not available on result type
        // skipped: field 'batch.total_count' not available on result type
    }

    @Test
    void testScrapeBatchProgress() throws Exception {
        // Batch scrape results include specific URL
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'batch.total_count' not available on result type
        // skipped: field 'batch.results' not available on result type
    }

}

File: e2e/rust/tests/batch_test.rs
//! E2e tests for category: batch

use kreuzcrawl::scrape;
use kreuzcrawl::create_engine;

#[tokio::test]
async fn test_scrape_batch_basic() {
    // Batch scrape of multiple URLs all succeeding
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'batch.completed_count' not available on result type
    // skipped: field 'batch.failed_count' not available on result type
    // skipped: field 'batch.total_count' not available on result type
}

#[tokio::test]
async fn test_scrape_batch_partial_failure() {
    // Batch scrape with one URL failing returns partial results
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'batch.completed_count' not available on result type
    // skipped: field 'batch.failed_count' not available on result type
    // skipped: field 'batch.total_count' not available on result type
}

#[tokio::test]
async fn test_scrape_batch_progress() {
    // Batch scrape results include specific URL
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'batch.total_count' not available on result type
    // skipped: field 'batch.results' not available on result type
}
